code/game/objects/structures/bedsheet_bin.dm

```diff
@@ -127,6 +175,70 @@ LINEN BINS
 /obj/item/weapon/bedsheet/brown/cargo
 	_color = "cargo"		//exists for washing machines, is not different from brown bedsheet in any way
 
+/obj/item/weapon/bedsheet/dye_act(var/obj/structure/reagent_dispensers/cauldron/cauldron, var/mob/user)
+	to_chat(user, "<span class='notice'>You begin dyeing \the [src].</span>")
+	playsound(cauldron.loc, 'sound/effects/slosh.ogg', 25, 1)
+	if (do_after(user, cauldron, 30))
+		var/mixed_color = mix_color_from_reagents(cauldron.reagents.reagent_list, TRUE)
+		var/mixed_alpha = mix_alpha_from_reagents(cauldron.reagents.reagent_list)
+		if (copytext(icon_state, 1, 6) == "plaid")
+			icon_state = "plaidsheetdyeable"
+			color = BlendRGB(color, mixed_color, mixed_alpha/255)
+		else
+			overlays.len = 0
+			icon_state = "sheetdyeable"
+			color = BlendRGB(color, mixed_color, mixed_alpha/255)
+			var/image/I = image(icon, src, "sheet-overlay")
+			I.appearance_flags = RESET_COLOR
+			overlays += I
+			update_blood_overlay()
+		user.update_inv_hands()
+	return TRUE
+
+/obj/item/weapon/bedsheet/proc/plaid_convert()
+	switch(icon_state)
```

Comment:
uh.... how about if(icon_state) in valid_colors then  icon_state = "plaid" + icon_state?

Reply:
Yeah that's more elegant, I changed it to something like that.

I've tested it and it works both ways properly.